crates/uv/src/commands/project/run.rs

```diff
@@ -209,6 +240,47 @@ pub(crate) async fn run(
             }
         };
 
+        let project = if no_project {
+            // If the user runs with `--no-project` and we can't find a project, warn.
+            if project.is_none() {
+                warn_user_once!("`--no-project` was provided, but no project was found");
+            }
+
+            // If the user ran with `--no-project` and provided a project-only setting, warn.
+            if !extras.is_empty() {
+                warn_user_once!("Extras have no effect alongside `--no-project`");
+            }
+            if !dev {
+                warn_user_once!("`--no-dev` has no effect alongside `--no-project`");
+            }
+            if locked {
+                warn_user_once!("`--locked` has no effect alongside `--no-project`");
+            }
+            if frozen {
+                warn_user_once!("`--frozen` has no effect alongside `--no-project`");
+            }
```

Comment:
We could enforce these _specific_ warnings via Clap, and make them errors. But the others must be done at runtime, because we don't _know_ if we'll discover a project when we first parse the arguments. So I just did them all as warnings for consistency.